api/v1/categories/PkpCategoryCategoryController.php

```diff
@@ -0,0 +1,409 @@
+            }
+        }
+
+        $locales = $context->getSupportedFormLocaleNames();
+        $locales = array_map(fn (string $locale, string $name) => ['key' => $locale, 'label' => $name], array_keys($locales), $locales);
+        $publicFileManager = new PublicFileManager();
+        $baseUrl = $request->getBaseUrl() . '/' . $publicFileManager->getContextFilesPath($context->getId());
+        $temporaryFileApiUrl = $request->getDispatcher()->url($request, Application::ROUTE_API, $context->getPath(), 'temporaryFiles');
+
+        $categoriesApiUrl = $request->getDispatcher()->url(
+            Application::get()->getRequest(),
+            Application::ROUTE_API,
+            $context->getPath(),
+            'categories'
+        );
+
+        $form = new CategoryForm($categoriesApiUrl, $locales, $baseUrl, $temporaryFileApiUrl, $category ?: null);
+        return response()->json($form->getConfig(), Response::HTTP_OK);
+    }
+
+    /**
+     * Delete a category by ID.
+     */
+    public function delete(Request $illuminateRequest): JsonResponse
+    {
+        $categoryId = $illuminateRequest->route('categoryId');
+        $context = $this->getRequest()->getContext();
+
+        $category = Repo::category()->get($categoryId, $context->getId());
+        if (!$category) {
+            return response()->json(__('api.404.resourceNotFound'), Response::HTTP_NOT_FOUND);
+        }
+
+        $image = $category->getImage();
+        Repo::category()->delete($category);
+
+        if ($image) {
+            $publicFileManager = new PublicFileManager();
+            $publicFileManager->removeContextFile($category->getContextId(), $image['uploadName']);
+            $publicFileManager->removeContextFile($category->getContextId(), $image['thumbnailName']);
+        }
+
+        return response()->json([], Response::HTTP_OK);
+    }
+
+    /**
+     * Generate the thumbnail image when creating a category.
+     *
+     * @return array - assoc array with thumbnail details
+     * Example:
+     * ```
+     * [
+     *  'thumbnailName' => (string),
+     *  'thumbnailWidth' => (int),
+     *  'thumbnailHeight' => (int),
+     * ]
+     * ```
+     */
+    private function generateThumbnail(TemporaryFile $temporaryFile, Context $context, $categoryId): array
+    {
+        $temporaryFileManager = new TemporaryFileManager();
+        $imageExtension = $temporaryFileManager->getImageExtension($temporaryFile->getFileType());
+        $sizeArray = getimagesize($temporaryFile->getFilePath());
+        $temporaryFilePath = $temporaryFile->getFilePath();
+
+        // Generate the surrogate images. Used later to create thumbnail
+        $image = match ($imageExtension) {
+            '.jpg' => imagecreatefromjpeg($temporaryFilePath),
+            '.png' => imagecreatefrompng($temporaryFilePath),
+            '.gif' => imagecreatefromgif($temporaryFilePath),
+        };
```

Comment:
I'm not sure which image formats we're supporting, but if the user uploads something unexpected, an exception should be raised here. Do we have a validation (e.g. "only jpg/png/gif are allowed")?

Reply:
Yea, a validation is done earlier before this function call